docs/spec/spec.md

```diff
@@ -0,0 +1,309 @@
+# Object Model
+
+This is the detailed specification of the Knative Eventing Object Model. For
+context, see the [overview](overview.md), [motivation](motivation.md), and the
+[interface contract](interfaces.md) documents.
+
+ * [Source](#kind-source) 
+ * [Channel](#kind-channel)
+ * [Subscription](#kind-subscription)
+ * [Provider](#kind-provisioner)
+
+---
+
+## kind: Source
+
+### group: eventing.knative.dev/v1alpha1
+
+_Describes a specific configuration (credentials, etc) of a source system which
+can be used to supply events. Sources emit events using a channel specified in
+their status. They cannot receive events._
+
+### Object Schema
+
+#### Spec
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| provisioner*| ProvisionerReference | The provisioner used to create any backing resources and configuration. | Immutable. |
+| arguments | runtime.RawExtension (JSON object)| Arguments passed to the provisioner for this specific source. | Arguments must validate against provisioner's parameters. |
+| channel | ObjectRef | Specify an existing channel to use to emit events. If empty, create a new Channel using the cluster/namespace default. | Source will not emit events until channel exists. |
+
+\*: Required
+
+#### Status
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| subscribable | Subscribable | Pointer to a channel which can be subscribed to in order to receive events from this source. | |
+| provisioned |[]ProvisionedObjectStatus| Creation status of each Channel and errors therein. | It is expected that a Source list all produced Channels. |
+
+##### Conditions
+
+ * **Ready.** True when the Source is provisioned and ready to emit events.
+ * **Provisioned.** True when the Source has been provisioned by a controller.
+
+#### Events
+
+ * Provisioned - describes each resource that is provisioned.
+
+### Life Cycle
```

Comment:
I don't understand what "Life Cycle" is for.

Reply:
It is suppose to be the reactions that the controller will do on the verbs `create` `update` and `delete`.

Not useful?